Model/ModelManager.php

```diff
@@ -385,8 +386,12 @@ public function getNormalizedIdentifier($entity)
             throw new \RunTimeException('Invalid argument, object or null required');
         }
 
-        // the entities is not managed
-        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
+        if (!$entity) {
+            return;
+        }
+
+        $entityState = $this->getEntityManager($entity)->getUnitOfWork()->getEntityState($entity);
+        if (UnitOfWork::STATE_NEW == $entityState || UnitOfWork::STATE_REMOVED == $entityState) {
```

Comment:
> would we really care if it returns '2' as a string for some reason instead of 2 as an int? (or '4')?

Yes we really care of this. If it returns integers, let's compare integers.

Strict comparison is more reliable. For example, if this method returns also a boolean for some reasons on a next releases, this can have an impact on our comparison if not strict.

So if we can do a strict comparison, we have to do a strict comparison. :wink: 

For the `in_array` usage, I never been told about performance or anything else. But I would love to read an article about this if you have it!